File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int l=lowerbound(nums,target);
        if(l==nums.size() || nums[l]!=target) return {-1,-1};
        int r=lowerbound(nums,target+1)-1;
        return {l,r};
    }
    int lowerbound(vector<int>& nums,int x){
        int l=0,r=nums.size();
        while(l<r){
            int mid=(l+r)/2;
            if(nums[mid]>=x) r=mid;
            else l=mid+1;
        }
        return l;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
def lowerbound(nums:List[int],target:int)->int:
    l=-1
    r=len(nums)
    while l+1<r:
        mid=(l+r)//2
        if nums[mid]>=target:
            r=mid
        else:
            l=mid
    return r
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        start=lowerbound(nums,target)
        if start==len(nums) or nums[start]!=target:
            return [-1,-1]
        end=lowerbound(nums,target+1)-1
        return [start,end]